libkbfs/folder_block_ops.go

```diff
@@ -1299,8 +1299,8 @@ func (fbo *folderBlockOps) writeDataLocked(
 			de.EncodedSize = 0
 			// update the file info
 			de.Size += uint64(len(block.Contents) - oldLen)
-			fbo.deCache[file.tailPointer().ref()] = de
 		}
+		fbo.deCache[file.tailPointer().ref()] = de
```

Comment:
Ah right, maybe this deserves its own PR.  In the current code, the file only gets added to the `deCache` when it is extended.  If some earlier bytes are overwritten, it won't be put into `deCache`.  And `deCache` is what's used to tell if a file (and the TLF in general) has any dirty bytes!  So if some earlier bytes were overwritten, the file would never look dirty, and the background sync would never kick in.